server/src/main/java/org/elasticsearch/index/query/MatchPhraseQueryBuilder.java

```diff
@@ -172,11 +173,8 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
             throw new QueryShardException(context, "[" + NAME + "] analyzer [" + analyzer + "] not found");
         }
 
-        MatchQuery matchQuery = new MatchQuery(context);
-        if (analyzer != null) {
-            matchQuery.setAnalyzer(analyzer);
-        }
-        matchQuery.setPhraseSlop(slop);
+        MappedFieldType fieldType = context.fieldMapper(fieldName);
+        MatchQuery matchQuery = fieldType.matchQuery(context, analyzer, slop);
```

Comment:
Will the same work for `multi_match`, `query_string` and `simple_query_string` ? I don't think you need to override `match_query`, calling `MappedFieldType#analyzePhrase` should be enough ? There's also an issue for query time synonyms where we use span queries instead of phrases but this can be done in a follow up.